src/main/java/de/tum/in/www1/artemis/web/rest/AthenaResource.java

```diff
@@ -140,6 +148,44 @@ public ResponseEntity<List<ProgrammingFeedbackDTO>> getProgrammingFeedbackSugges
                 athenaFeedbackSuggestionsService::getProgrammingFeedbackSuggestions);
     }
 
+    /**
+     * GET athena/modeling-exercises/:exerciseId/submissions/:submissionId/feedback-suggestions : Get feedback suggestions from Athena for a modeling exercise
+     *
+     * @param exerciseId   the id of the exercise the submission belongs to
+     * @param submissionId the id of the submission to get feedback suggestions for
+     * @return 200 Ok if successful with the corresponding result as body
+     */
+    @GetMapping("athena/modeling-exercises/{exerciseId}/submissions/{submissionId}/feedback-suggestions")
+    @EnforceAtLeastTutor
+    public ResponseEntity<List<ModelingFeedbackDTO>> getModelingFeedbackSuggestions(@PathVariable long exerciseId, @PathVariable long submissionId) {
+        return getFeedbackSuggestions(exerciseId, submissionId, modelingExerciseRepository::findByIdElseThrow, modelingSubmissionRepository::findByIdElseThrow,
```

Comment:
this code uses the convention for other exercise types so this comment actually relates to existing code rather than the added code here, and perhaps this comment can be ignored. that said:

- this method of fetching feedback for a particular submission seems a bit inefficient. couldn't this be done with one query instead of multiple independent queries?
- the `getFeedbackSuggestions()` method doesn't seem to bundle much functionality beyond running queries subsequently, a log and an auth check, which to me looks it makes the code unnecessarily confusing without any tangible benefits.

Reply:
Fair point - I will talk to Stephan if it makes sense to generally refactor the structure here in a follow up 🤔